h2o-automl/src/test/java/ai/h2o/automl/targetencoding/TEMojoIntegrationTest.java

```diff
@@ -0,0 +1,547 @@
+
+      asFactor(fr, responseColumnName);
+
+      Scope.track(fr);
+
+      String[] teColumns = {"home.dest", "embarked"};
+
+      TargetEncoderModel.TargetEncoderParameters targetEncoderParameters = new TargetEncoderModel.TargetEncoderParameters();
+      targetEncoderParameters._withBlending = false;
+      targetEncoderParameters._columnNamesToEncode = teColumns;
+      targetEncoderParameters._ignore_const_cols = false; // Why ignore_const_column ignores `name` column? bad naming
+      targetEncoderParameters.setTrain(fr._key);
+      targetEncoderParameters._response_column = responseColumnName;
+
+      TargetEncoderBuilder targetEncoderBuilder = new TargetEncoderBuilder(targetEncoderParameters);
+
+      targetEncoderBuilder.trainModel().get();
+
+      targetEncoderModel = targetEncoderBuilder.getTargetEncoderModel();
+
+      try (FileOutputStream modelOutput = new FileOutputStream(mojoFile)) {
+        targetEncoderModel.getMojo().writeTo(modelOutput);
+        System.out.println("Model has been written down to a file as a mojo: " + mojoFileName);
+      }
+
+      // Let's load model that we just have written and use it for prediction.
+      EasyPredictModelWrapper teModelWrapper = null;
+
+      TargetEncoderMojoModel loadedMojoModel = (TargetEncoderMojoModel) MojoModel.load(mojoFile.getPath());
+
+      teModelWrapper = new EasyPredictModelWrapper(loadedMojoModel);
+
+      // RowData that is not encoded yet
+      RowData rowToPredictFor = new RowData();
+      String homeDestFactorValue = "Montreal  PQ / Chesterville  ON";
+      String embarkedFactorValue = "S";
+
+      rowToPredictFor.put("home.dest", homeDestFactorValue);
+      rowToPredictFor.put("embarked", embarkedFactorValue);
+
+      double[] currentEncodings = teModelWrapper.transformWithTargetEncoding(rowToPredictFor).transformations;
+      //Let's check that specified in the test categorical columns have been encoded in accordance with targetEncodingMap
+      EncodingMaps targetEncodingMap = loadedMojoModel._targetEncodingMap;
+
+      double encodingForHomeDest = checkEncodingsByFactorValue(fr, homeDestFactorValue, targetEncodingMap, "home.dest");
+
+      double encodingForHomeEmbarked = checkEncodingsByFactorValue(fr, embarkedFactorValue, targetEncodingMap, "embarked");
+
+      // Because of the random swap we need to know which index is lower so that we know order of transformations/predictions
+      int currentHomeDestPredIdx = fr.find("home.dest") < fr.find("embarked") ? 0 : 1;
+
+      assertEquals(currentEncodings[currentHomeDestPredIdx], encodingForHomeDest, 1e-5);
+      assertEquals(currentEncodings[currentHomeDestPredIdx == 0 ? 1 : 0], encodingForHomeEmbarked, 1e-5);
+    } finally {
+      targetEncoderModel.remove();
+      Scope.exit();
+    }
+  }
+
+  private void compareWithReference(double[] encodings, int homeDestPredIdx, int swapIdx1, int swapIdx2, double[] currentEncodings, int currentHomeDestPredIdx) {
```

Comment:
I don't like this - it hides the fact that the test is actually not testing anything, if a test is unstable we should just mark it @Ignore and fix in next PR

Reply:
@honzasterba I don't agree. Please read the name of the test - it explicitly says that it test consistency and does not hide anything. Tests could test performance, reliability(stress testing), consistency, whatever and keep us safe from regression in a long run.  Let's imagine we follow your suggestion - we put Ignore annotation and then address the issue in another PR. How would we then test and assure themselves that issue is solved in this separate PR - right, with test that checks consistency (remember you said it tests nothing). 
Please keep in mind that people might have reasons to step back from conventions even if they know them. My goal was to attract attention from others that TE functionality relies on other functionality which is unstable. In general I ,of course, don't want to see how my builds are failing but I don't want to hide important concerns. Before merging it I would put ignore annotation (PR already exist).
Another point that I wanted to discuss (cc @michalkurka @seb-h2o ) is  NOPASS mechanism (which supports only Py and maybe R) which is designed for flaky tests. If we knew conditions under which those are failing we would be able to use concept of Assume from JUnit - but because we have NOPASS looks like we either don't know the issue in the test or we don't have Assume concept at hand. Question is : are we using NOPASS for unstable - due to unknown issues in test - tests? If answer is NO then maybe we should use Assume concept instead of NOPASS?

Reply:
ok, the tests actuall tests something
but this method only does Log.warn in worst case, so the whole assert..catch brings no value and the same can be done with a simple if
hoever a tests either passes of fails, the side effect of printing something to the output is usually lost and not paid attention to, in which case I would remove this because it does have no impact on the result of the test

Reply:
@honzasterba I agree with concern that logs are not that noticeable. The reason for catching assertions is that I wanted to show how frequently test is failing. Also this test was extracted out from one test that was checking two things. I decided to move consistency part into a dedicated test to make it more explicit in terms of code. Probably I should have made it to fail more explicit being in a separate test. Anyway I was planning to add ignore annotation and make it fail with exceptions before merging it. Next time I will add comment or TODO so that my intentions are more clear. Thanks for pointing out to this!